js/front/dataset/resource/availability.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+    <span v-if="checkAvailability" :class="['label', checkAvailability.class]">{{ checkAvailability.message }}</span>
+</template>
+
+<script>
+export default {
+    name: 'availability',
+    props: {
+        /**
+         * Resource object as extracted from JSON-LD
+         */
+        resource: {
+            type: Object,
+        },
+        /**
+         * The url to get a refreshed check status from
+         */
+        checkurl: {
+            type: String,
+        },
+        /**
+         * A status to be used directly instead of parsing the resource and
+         * calling checkurl
+         */
+        status: {
+            type: Number,
+        }
+    },
+    data() {
+        return {
+            checkStatus: undefined,
+        }
+    },
+    created() {
+        if (this.status) {
+            this.checkStatus = this.status;
+            return;
+        }
+        this.getResourceCheckStatus(this.resource, this.checkurl)
+            .then((res) => {
+                this.checkStatus = res['check:status'];
+            })
+            .catch(error => {
+                console.log('Something went wrong with the linkchecker', error);
+            });
+    },
+    methods: {
+        /**
+         * Get cached check or API check
+         * @param  {Object} resource A resource element from DOM
+         * @param  {String} checkurl The API check url
+         */
+        getResourceCheckStatus(resource, checkurl) {
+            const cachedCheck = this.getCachedCheck(resource);
+            return (cachedCheck && Promise.resolve(cachedCheck)) || this.$api.get(checkurl);
+        },
+        /**
+         * Get a cached checked result from extras if resource is not flagged
+         * as needing a new check
+         * @param  {Object} resource A resource as extracted from JSON-LD
+         */
+        getCachedCheck(resource) {
+            if (!resource.needCheck) {
+                const extras = this.getCheckExtras(resource.extras || []);
+                if (extras['check:status']) {
+                    return extras;
+                }
+            }
+        },
+        /**
+         * Get check related extras from JSON-LD
+         * @param {Array} extras A list of extras in JSON-LD format
+         * @return {Object} Check extras as a hash, if any
+         */
+        getCheckExtras(extras) {
+            return extras.reduce((obj, extra) => {
+                if (extra.name.startsWith('check:')) {
+                    obj[extra.name] = extra.value;
+                }
+                return obj;
+            }, {});
+        },
+    },
+    computed:{
```

Comment:
Put `computed` between `props` and `data` to keep the same order as other components